# Sam Gurr
# 2021 Self-Assessment
# Goals this year 2021
---
layout: post
title: Self_Assessment_Plan
date: '2021-03-12'
categories: Processing
tags: Self, Goals
---

## (1)	RESEARCH AND WRITING GOALS

---------------------------------- 


## I.	Manuscripts


### i.	Phys repeat exposure project: 
 -	resubmit to JEB (deadline March 5th)
 -	pending second comments – edit and submit (fingers crossed)
 
 
### ii.	TagSeq project (if expression data only)
 -	Finalize all analysis tables and main figures by April 1st
 -	Draft manuscript by June 1st 2021
 -	Note: add more metrics here – metabolomics/seq data for insight on non-genetic transcriptional regulation?


## II.	Experiment(s) and Measurements  

---------------------------------- 

### i.	TagSeq project: 
 -	Metabolomics: 
	-	Homogenize geoduck for metabolomics (supplement TagSeq data). Learn KEGG or MAGI to integrate metabolomics with Different Expression data. 
 -	DNA seq ideas: 
	-	Sequence DNA to compliment TagSeq data - Integrate a multifaceted molecular approach – strengthen findings from mRNA results with non-genetic insight on transcriptional regulation (i.e. methylation, chromatin accessibility - may not be feasible in dissertation timeline). 
	-	Note: Free RRBS kit from Zymo still available in lab. 
	
**Rational / Hypothesis for expansion:** 
	
a. Rationale: From TagSeq data we see the weight of expression trajectory favors higher transcription in the 'ambient' control animals.
	  We also see subsequent encounters elicit complex patterns of expression dependent on stress intensity and conditioning. 
	  In particular, GO analysis following WGCNA show evidence that transcription post-secondary treatment ( Days 7 and 14 )
	  upregulated genes asspcaoited with oxidative stess, methanionine cycle (transport of SAM), and ferrix iron transport.
	  Day14_WGCNA

DAY 7 WGCNA - module brown highlights primary treatment effect ambient > moderate (vst transformed expression) 

![Day7_WGCNA](https://samgurr.github.io/SamJGurr_Lab_Notebook/images/Day7_WGCNA.jpg "Day7_WGCNA")

DAY 7 goseq followup on brown module -  evidence of oxidative stress  + methanionine cycle + ferrous/feric iron pathways
  
![WGCNA_Day7_moduleBrown_goseq](https://samgurr.github.io/SamJGurr_Lab_Notebook/images/WGCNA_Day7_moduleBrown_goseq.JPG "WGCNA_Day7_moduleBrown_goseq")



DAY 14 WGCNA - module brown highlights primary treatment effect ambient > moderate (vst transformed expression) 

![Day14_WGCNA](https://samgurr.github.io/SamJGurr_Lab_Notebook/images/Day14_WGCNA.jpg "Day14_WGCNA")

DAY 14 goseq followup on brown module -  evidence of oxidative stress  + methanionine cycle + ferrous/feric iron pathways
  
![WGCNA_Day14_moduleBrown_goseq](https://samgurr.github.io/SamJGurr_Lab_Notebook/images/WGCNA_Day14_moduleBrown_goseq.JPG "WGCNA_Day14_moduleBrown_goseq")




 - **Stress conditioning** (OA induced ROS and/or caloric restriction) --> 
	  low glycolysis --> 
	  dependence on OXPHOS + ROS-induced reverse electron transport (RET) --> 
	  high NAD --> 
	  sirtuin activation --> 
	  deacetylation of proteins in cytosol (activating them) + deaceylation of histones (**histone condensation**) -->
      **regulated transcription and decrease in spurious transcription**  
	  
	  - Why is the *moderate conditioned* phenotype attributed with lower expression of methanione cycle, oxidative stress response, and ferric iron transport? 
	  This is possibly the result of (1) higher NAD pool during early life conditioning phase (2) active sirtuins (3) "focused" transcriptional regulation (histones condenced) 
      Perhaps the 'ambient' phenotype is pushing for transcriptional control via methylation?   
	  
	  b. Hypothesis: 
		- Early stress acclimation increases the NAD pool enhancing sirtuin-medaited deacetylation of histones reducing spurious transciption under subseqent stress encounters (we found downregulated geneExp in moderate conditioned animals). 
	    - Animals without prior conditioning have open chromatin regions and depend on methyl-transport to regulate transcription during stress encounters. 

![OA_mechanism_flowchart](https://samgurr.github.io/SamJGurr_Lab_Notebook/images/OA_mechanism_flowchart.JPG "OA_mechanism_flowchart")


![2020_AOX_RET_Crosstalk](https://samgurr.github.io/SamJGurr_Lab_Notebook/images/2020_AOX_RET_Crosstalk.JPG "2020_AOX_RET_Crosstalk")


### ii. Cross-generation, dig out-planted geoduck. 

About: geoduck from two OA-treated parental histories (ambient and tremant) successively 
spawned simultaneously and were raised separately throughout the industry rearing process in 2020. 
Juveniles were outplanted in Joth Davis' mudflat and we talking about a summer dig to study these animals.

 -	Immedaite upon dig...
	-	Measure phenotype:
		a. *Simple approach*: Shell size, organic biomass (AFDW), and respiration rate
	    b. *Complex approach*: P = (C x A) – R +E.  --> Growth = [ energy absorbed – (energy respired + excretion) ]
	-	Sequencing 
		a. *N* = 4 animals per parental treatment
 -	Acute exposure trials...
	-  Experimental treatment:
		a. Proposed NSF postdoc experiment - multistressor + inhibitor (suramin, SHAM, and/or DNMT inhibitor) 
    -  Measurements: 
		a. Sequencing (gill and muscle tissues) - TagSeq + methyl-ATACseq or other? 
		
## III.	Overall Goals This Year..

(1)	Bioinformatics skills
i. Gain experience using bioinformatic tools with diverse sequence datasets. 
i.	Interpret molecualr data and write a manuscript sumamrizing the findings using this new array of skills
i.  Potentially integrate a multifaceted sequencing approach in my dissertation – supplement findings from TagSeq with non-genetic insight on transcriptional regulation (i.e. methylation, chromatin accessibility - may not be feasible in dissertation timeline)
NOTE: I recently finalized two post-doc applications proposing work with ddRADseq and methyl-ATAC-seq. I also have a free RRBS kit from Zymo in Putnam Lab that I could give a try funds and time permitting.

(2)	Apply to postdocs / positions 
a.	Applied for NOAA NRC and NSF fellowships 
b.	Search more possibilities and continue the conversation (PIs of interest for postdocs, etc.)

(3)	Defend 
a.	Plan to defend in late winter 2021

(5)	Manuscript publication
a.	JEB manuscript 
b.	Differential expression manuscript (with possible metabolomics and non-genetic transcriptional regulation data)
c.	Draft results of the acute outplant experiment (planning for summer 2021)

